types/protoo-server/index.d.ts

```diff
@@ -0,0 +1,92 @@
+// Type definitions for protoo-server 4.0
+// Project: https://protoojs.org
+// Definitions by: Antonis Balasas <https://github.com/antoniom>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { Server as HttpServer, IncomingMessage } from 'http';
+import { Server as HttpsServer } from 'https';
+import { Socket } from 'net';
+import { IServerConfig } from 'websocket';
+
+export interface ConnectionRequestInfo {
+    request: IncomingMessage;
+    origin: string;
+    socket: Socket;
+}
+
+export interface ProtooRequest {
+    request: true;
+    id: number;
+    method: string;
+    data: any;
+}
+
+export interface ProotooResponse {
+    response: true;
+    id: number;
+    ok: boolean;
+    data?: any;
+    errorCode?: number;
+    errorReason?: string;
+}
+
+export interface ProtooNotification {
+    notification: true;
+    method: string;
+    data: any;
+}
+
+export type ConnectionRequestCb = (
+    info: ConnectionRequestInfo,
+    accept: ConnectionRequestAcceptFn,
+    reject: ConnectionRequestRejectFn,
+) => void;
+
+export type ConnectionRequestAcceptFn = () => WebSocketTransport;
+
+export type ConnectionRequestRejectFn = ((code: number, reason: string) => void) | ((error: Error) => void);
+
+export type RequestCb = (request: ProtooRequest, accept: AcceptFn, reject: RejectFn) => void;
+
+export type AcceptFn = (data: any) => void;
+
+export type RejectFn = ((errorCode?: Error) => void) | ((errorCode: number, errorReason: Error | string) => void);
+
+export type EmptyCb = () => void;
+
+export type NotificationCb = (notification: ProtooNotification) => void;
+
+export class WebSocketServer {
+    constructor(server: HttpServer | HttpsServer, options?: IServerConfig);
+    stop(): void;
+    on(eventType: 'connectionrequest', callback: ConnectionRequestCb): void;
+}
+
+export class WebSocketTransport {}
```

Comment:
This definition cannot be present here since this library does not export anything like this.
```js
const m = require("protoo-server");
console.log(m.WebSocketTransport );
```
this reports `undefined`